module3/Homework3Playground.playground/Contents.swift

```diff
@@ -0,0 +1,175 @@
+
+
+//a) Create an array of Int called nums with values 0 through 20. Iterate over the Array and print the even numbers.
+print("a) ---------------")
+let firstArry = Array(0...20)
+for num in firstArry where num % 2 == 0 {
+  print(num)
+}
+
+//b) In your Playground, define the following String:
+//
+//let sentence = “The qUIck bRown fOx jumpEd over the lAzy doG”
+//
+//Iterate over sentence counting the vowels (a, e, i, o, u), ignoring the case.
+
+print("b) ---------------")
+let sentence = "The qUIck bRown fOx jumpEd over the lAzy doG"
+var vowelCount = 0
+let lowercasedSentence = sentence.lowercased()
+let vowels = Set(["a", "e", "i", "o", "u"])
+for character in lowercasedSentence where vowels.contains(String(character)) {
+  vowelCount += 1
+}
+print(vowelCount)
+
+
+//c) Create two arrays of Int with the values of 0 through 4. Use a nested for loop to print a mini multiplication table. The output, which should be multiple lines, should be in the following format:
+//
+//0 * 0 = 0
+//
+//
+//The format follows the number from the first array, space, followed by the character *, space,  followed by the number from the second array, space,  followed by =, space, followed by the result of the number from the first array multiplied by the number from the second array (just like the example above). There are a couple of ways to achieve this, but String Interpolation is the easiest.
+
+print("c) ---------------")
+let arrayA = Array(1...4)
+let arrayB = Array(1...4)
+
+for numberA in arrayA {
+  for numberB in arrayB {
+    print("\(numberA) * \(numberB) = \(numberA * numberB)")
+  }
+}
+
+//d) Write a function called average that takes an optional array of Int. If the array is not nil, calculate the average of the array's values and print:
+//
+//“The average of the values in the array is <average>.”
+//
+//
+//Where <average> is the calculated average. If the array is nil, print:
+//
+//“The array is nil. Calculating the average is impossible.”
+//
+//
+//Note: the average is calculated by summing the values in the array and dividing by the number of elements.
+//
+//
+//Call this function two times. First, pass in the nums array from part A and second by passing an optional array of Int.
+
+print("d) ---------------")
+
+func average(_ array: [Int]?) {
+  if let array = array {
+    var sum = 0
+    for el in array {
```

Comment:
I'll have to go look to see if they are introduced yet, but there are some nice methods on collection types of `map`, `flatMap`, `reduce` and others. This would be a good time to use `reduce`:
````
let sum = array.reduce(0,+)
````